src/extension.ts

```diff
@@ -62,10 +62,17 @@ export function activate(context: ExtensionContext) {
 
     // Register support for unicode input.
     (async () => {
-        const translations: any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const translations : any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const modes = [LEAN_MODE];
+        const input_markdown = workspace.getConfiguration('lean.input').get('markdown');
+        if (input_markdown) {modes.push(MARKDOWN_MODE);}
```

Comment:
Please move the code that reads the desired modes from the configuration to `input.ts`.  Also make sure that the modes are reread on configuration change events (there are already event listeners installed).

Look at `inputModeLeader()` for inspiration.

Reply:
done